### PR TITLE
PMM-7874 UI tests migration

### PR DESCRIPTION
**Pull request by yurkovychv**

### CODE REVIEW

File: pmm/pmm2-ui-tests.groovy [hunk 1]
```diff
@@ -158,15 +158,15 @@ pipeline {
     options {
         skipDefaultCheckout()
     }
-    triggers {
-        upstream upstreamProjects: 'pmm2-server-autobuild', threshold: hudson.model.Result.SUCCESS
-    }
+   // triggers {
```
Comment (puneet0191) [inline pmm/pmm2-ui-tests.groovy:161]: Remove comments here

File: pmm/pmm2-upgrade-tests.groovy [hunk 2]
```diff
@@ -203,35 +199,35 @@ pipeline {
...
-                    junit 'pmm-app/tests/output/parallel_chunk*/*.xml'
+                    junit 'tests/output/parallel_chunk*/*.xml'
                     slackSend channel: '#pmm-ci', color: '#FF0000', message: "[${JOB_NAME}]: build ${currentBuild.result} - ${BUILD_URL}"
-                    archiveArtifacts artifacts: 'pmm-app/tests/output/combine_results.html'
+                    archiveArtifacts artifacts: 'tests/output/combine_results.html'
                     archiveArtifacts artifacts: 'logs.zip'
-                    archiveArtifacts artifacts: 'pmm-app/tests/output/parallel_chunk*/*.png'
-                    archiveArtifacts artifacts: 'pmm-app/tests/output/video/*.mp4'
+                    archiveArtifacts artifacts: 'tests/output/parallel_chunk*/*.png'
+                    archiveArtifacts artifacts: 'tests/output/video/*.mp4'
```
Comment (puneet0191) [inline pmm/pmm2-upgrade-tests.groovy:218]: Remove this also, we don't have any video artifacts any more.

File: pmm/pmm2-upgrade-tests.groovy [hunk 3]
```diff
@@ -203,35 +199,35 @@ pipeline {
             '''
             destroyStaging(VM_NAME)
             sh '''
-                sudo chmod 777 -R pmm-app/tests/output
-                ./pmm-app/node_modules/.bin/mochawesome-merge pmm-app/tests/output/parallel_chunk*/*.json > pmm-app/tests/output/combine_results.json
-                ./pmm-app/node_modules/.bin/marge pmm-app/tests/output/combine_results.json --reportDir pmm-app/tests/output/ --inline --cdn --charts
+                sudo chmod 777 -R tests/output
```
Comment (puneet0191) [inline pmm/pmm2-upgrade-tests.groovy:202]: Make a false pass for this part, incase of any failure during setup this will not error out and crash the pipeline execution, `|| true` just like we do on nightly and UI tests.

**Review by puneet0191 — CHANGES_REQUESTED** [review]

File: pmm/pmm2-upgrade-tests.groovy (hunk 3 above)
Comment (yurkovychv) [reply pmm/pmm2-upgrade-tests.groovy]: done

File: pmm/pmm2-upgrade-tests.groovy (hunk 2 above)
Comment (yurkovychv) [reply pmm/pmm2-upgrade-tests.groovy]: done

File: pmm/pmm2-ui-tests.groovy (hunk 1 above)
Comment (yurkovychv) [reply pmm/pmm2-ui-tests.groovy]: done

File: pmm/pmm2-upgrade-tests.groovy [hunk 4]
```diff
@@ -199,39 +195,38 @@ pipeline {
...
             sh '''
-                curl --insecure ${PMM_URL}/logs.zip --output logs.zip
+                curl --insecure ${PMM_URL}/logs.zip --output logs.zip || true
             '''
             destroyStaging(VM_NAME)
             sh '''
-                sudo chmod 777 -R pmm-app/tests/output
-                ./pmm-app/node_modules/.bin/mochawesome-merge pmm-app/tests/output/parallel_chunk*/*.json > pmm-app/tests/output/combine_results.json
-                ./pmm-app/node_modules/.bin/marge pmm-app/tests/output/combine_results.json --reportDir pmm-app/tests/output/ --inline --cdn --charts
+                sudo chmod 777 -R tests/output || true
```
Comment (puneet0191) [inline pmm/pmm2-upgrade-tests.groovy:202]: I don't think we need this line anymore, because we don't use the docker container for running tests anymore, could you remove this and try testing it on any one of the pipeline.

**Review by puneet0191 — CHANGES_REQUESTED** [review]

**Review by atymchuk — CHANGES_REQUESTED** [review]
We need to test a scenario when tests are OK and tests/output is empty or doesn't exist (maybe creating an empty dir is simpler)

File: pmm/pmm2-ui-tests-nightly.groovy [hunk 5]
```diff
@@ -215,27 +215,19 @@ pipeline {
...
                     sudo yum install -y gettext
                     envsubst < env.list > env.generated.list
-                    popd
+                    
                 """
             }
         }
         stage('Sleep') {
             steps {
-                sh """
```
[the remark this answers is not in the document]
Comment (yurkovychv) [reply pmm/pmm2-ui-tests-nightly.groovy:231]: It's needed actually to start getting data for dashboards and QAN

File: pmm/pmm2-ui-tests-nightly.groovy [hunk 6]
```diff
@@ -265,9 +255,9 @@ pipeline {
             // stop staging
             sh '''
                 curl --insecure ${PMM_URL}/logs.zip --output logs.zip || true
-                sudo chmod 777 -R pmm-app/tests/output || true
-                ./pmm-app/node_modules/.bin/mochawesome-merge pmm-app/tests/output/parallel_chunk*/*.json > pmm-app/tests/output/combine_results.json || true
-                ./pmm-app/node_modules/.bin/marge pmm-app/tests/output/combine_results.json --reportDir pmm-app/tests/output/ --inline --cdn --charts || true
+                sudo chmod 777 -R tests/output || true
+                ./node_modules/.bin/mochawesome-merge tests/output/parallel_chunk*/*.json > tests/output/combine_results.json || true
```
[the remark this answers is not in the document]
Comment (yurkovychv) [reply pmm/pmm2-ui-tests-nightly.groovy:259]: it won't break because we have `|| true` condition here

File: pmm/pmm2-upgrade-tests.groovy (hunk 4 above)
Comment (yurkovychv) [reply pmm/pmm2-upgrade-tests.groovy:202]: should I try removing it in other pipelines?

**Review by puneet0191 — APPROVED** [review]

File: pmm/pmm2-ui-tests-nightly.groovy (hunk 6 above)
[the remark this answers is not in the document]
Comment (atymchuk) [reply pmm/pmm2-ui-tests-nightly.groovy:258]: > it won't break because we have `|| true` condition here

No, `chmod` wouldn't, but `mochawesome` could

**Review by atymchuk — APPROVED** [review]
